activerecord/lib/active_record/connection_adapters/postgresql/column.rb

```diff
@@ -6,15 +6,24 @@ module PostgreSQL
       class Column < ConnectionAdapters::Column # :nodoc:
         delegate :oid, :fmod, to: :sql_type_metadata
 
-        def initialize(*, serial: nil, **)
+        def initialize(*, serial: nil, generated: nil, **)
           super
           @serial = serial
+          @generated = generated
         end
 
         def serial?
           @serial
         end
 
+        def virtual?
+          @generated == "s"
```

Comment:
Should this be any non-nil value? Assuming `"s"` is specific to stored columns, anticipating that non-stored columns may be supported in PG 13+.

Reply:
It may be some time before virtual are supported. PG12 nor 13 support it.

The current `devel` documentation also states that only stored are supported: https://www.postgresql.org/docs/devel/ddl-generated-columns.html

Would something like this work?
```ruby
VIRTUAL_COLUMN_KEYS = %w[s].freeze # when PG supports virtual switch to %w[s v].freeze

# ...

def virtual?
  VIRTUAL_COLUMN_KEYS.include?(@generated)
end
```

Reply:
```ruby
# when PG supports virtual switch to %w[s v].freeze
```

This makes Rails brittle with respect to specific PostgreSQL versions supported, however.

Reply:
I see, so the alternative would be to check for a non-null non-empty string value. The presumption being that will be definitive no matter the value of the string.

Sounds good to me.